arrow-flight/src/sql/client.rs

```diff
@@ -110,6 +110,11 @@ impl FlightSqlServiceClient<Channel> {
         self.token = None;
     }
 
+    /// Share the bearer token with potentially different `DoGet` clients
+    pub fn token(&self) -> &Option<String> {
+        &self.token
+    }
```

Comment:
I think a more "ideomatic" API would return an Option, like:

```suggestion
    pub fn token(&self) -> Option<&String> {
        self.token.as_ref()
    }
```